geopandas/tools/overlay.py

```diff
@@ -220,16 +220,24 @@ def overlay(df1, df2, how="intersection", keep_geom_type=True):
             result = dfunion[dfunion["__idx1"].notnull()].copy()
 
     if keep_geom_type:
+        key_order = result.keys()
+        exploded = result.reset_index(drop=True).explode()
+        exploded.reset_index(level=0, inplace=True)
```

Comment:
Oh, do you know the reason for that? We certainly wouldn't want to use `inplace` on a frame passed in by a user, but I don't actually know a reason to avoid that for a frame created and used entirely within a geopandas function.

Reply:
Not really. I just have this in my head based on some code review comment. Might be due to consistency? Not sure. @jorisvandenbossche?